soroban-env-host/src/host.rs

```diff
@@ -704,11 +705,13 @@ impl Host {
             HostObject::BigInt(bi) => {
                 self.charge_budget(CostType::HostBigIntAllocCell, bi.bits() as u64)?;
             }
-            HostObject::ContractCode(cc) => {
-                if let ScContractCode::Wasm(c) = cc {
-                    self.charge_budget(CostType::HostContractCodeAllocCell, c.len() as u64)?;
+            HostObject::ContractCode(cc) => match cc {
+                ScContractCode::WasmRef(h) => {
```

Comment:
Fyi that clippy typically recommends using if let over match in situations where only a single case is required.

Reply:
This is actually completely gone now, so no need for the check.